packages/fury-adapter-oas3-parser/lib/parser/oas/parseServerVariablesArray.js

```diff
@@ -0,0 +1,27 @@
+const R = require('ramda');
+const pipeParseResult = require('../../pipeParseResult');
+const parseArray = require('../parseArray');
+const parseServerVariableObject = require('./parseServerVariableObject');
+
+const name = 'Server Variables Array';
+
+/**
+ * Parse Server Variable Array
+ *
+ * @param namespace {Namespace}
+ * @param element {Element}
+ * @returns ParseResult
+ *
+ * @private
+ */
+const parseServerVariablesArray = (context, element) => {
+  const { namespace } = context;
+
+  const parseServerVariables = pipeParseResult(namespace,
+    parseArray(context, name, R.curry(parseServerVariableObject)(context)),
+    array => new namespace.elements.HrefVariables(array.content));
+
+  return parseServerVariables(element);
+};
```

Comment:
I think perhaps you've mis-understood the OpenAPI spec as I don't think there is any server variables array defined. The `variables` property of "Server Object" contains an object (key-value) pair. For example:

```yaml

servers:
  - url: https://{username}.gigantic-server.com:{port}/{basePath}
    description: The production API server
    variables:
      username:
        # note! no enum here means it is an open value
        default: demo
        description: this value is assigned by the service provider, in this example `gigantic-server.com`
      port:
        enum:
          - '8443'
          - '443'
        default: '8443'
      basePath:
        # open meaning there is the opportunity to use special base paths as assigned by the provider, default is `v2`
        default: v2
```

"variables" contains an object with three keys "username", "port", and "basePath".

Reply:
`parseServerVariablesArray` should perhaps become `parseServerVariablesObject` which accepts an object such as:

```yaml
username:
  default: x
port:
  default '80'
```

and returns `HrefVariables`, such as:

```json
{
  "element": "hrefVariables",
  "content": [
    {
      "element": "member",
      "content": {
        "key": {
          "element": "string",
          "content": "username"
        },
        "value": { ... }
      }
    },
    {
      "element": "member",
      "content": {
        "key": {
          "element": "string",
          "content": "port"
        },
        "value": { ... }
      }
    }
  ]
}
```

Reply:
I see;
`serverVariableArray` was just a way to bring the server variables to the server object